### ISSUE TITLE
New lint: `empty_expect` - detect `.expect("")` with empty message

### ISSUE THREAD

**Issue by konard** [rust-lang/rust-clippy#16764]
## What it does

Detects calls to `.expect("")` where the expect message is an empty string literal.

## Why

When `clippy::unwrap_used` is denied, developers sometimes use `.expect("")` as a workaround — which provides no more context than `.unwrap()` on panic. The expect message should be meaningful.

## Example

```rust
// Bad - should lint
let x: Option<i32> = Some(1);
let _ = x.expect("");

let v: Result<i32, &str> = Ok(1);
let _ = v.expect("");
```

```rust
// Good - no lint
let _ = x.expect("value should be present after initialization");
let _ = v.expect("parsing always succeeds for valid input");
```

## Category

`style` or `suspicious`

## Real-world motivation

In our codebase we deny `unwrap_used` to enforce better error context. Some developers bypass this by writing `.expect("")`, which defeats the purpose. Currently there is no Clippy lint to catch this — we resort to `grep`-based CI checks.

**Comment by xtqqczze** [comment]
This lint doesn’t add much value and feels like policy enforcement rather than a general improvement:

- `.expect("")` is already an explicit choice and not inherently wrong
- it also doesn’t guarantee better outcomes: requiring a non-empty string just encourages placeholder messages like "error", which don’t improve diagnostics
- if a project wants to forbid this, it’s better handled in CI or with a custom lint rather than in Clippy

**Comment by ada4a** [comment]
> Some developers bypass this by writing `.expect("")`, which defeats the purpose

This seems to hint at some deeper cultural issue? I agree with the previous commenter in that this doesn't sound like something clippy can fix for you.
